src/tensorflow.cc

```diff
@@ -1280,31 +1283,55 @@ AutoCompleteHelper::FixBatchingSupport()
               } else {
                 config_io.MemberAsArray("dims", &config_dims);
               }
-              if (config_dims.ArraySize() != 0) {
-                // look up corresponding io info from model
-                for (const TRITONTF_IOList* itr = model_ios[ios_idx];
-                     itr != nullptr; itr = itr->next_) {
-                  TRITONTF_IO* io = itr->io_;
-                  if (config_name == io->name_) {
-                    bool should_batch =
-                        (io->shape_->rank_ == (config_dims.ArraySize() + 1));
-                    // inconsistent hint
-                    if (config_batch_hint &&
-                        (model_support_batching_ != should_batch)) {
-                      return TRITONSERVER_ErrorNew(
+              for (const TRITONTF_IOList* itr = model_ios[ios_idx];
+                   itr != nullptr; itr = itr->next_) {
+                TRITONTF_IO* io = itr->io_;
+                if (config_name == io->name_) {
+                  bool model_io_explicit = io->shape_->rank_ > 0;
+                  bool user_config_is_defined = config_dims.ArraySize() > 0;
+
+                  if (model_io_explicit && !user_config_is_defined) {
+                    config_batch_hint = true;
```

Comment:
I think a lack of user provided config tells us nothing. Why are you setting the variable true here?

Reply:
Never mind. This is within `sig_supports_batch` part.

Reply:
L1211 `bool sig_supports_batch = ModelSupportsBatch(model_ios);` does an early out check to see if the model can support batching. If we have made it to this conditional then the saved model supports batching but we have no confirmation from the user config; thus, we set the hint to true.